### PR TITLE
Adds margin_ranking_loss opinfo

### PR DESCRIPTION
**Pull request by mruberry**
per title

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -11796,6 +11814,20 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
                DecorateInfo(unittest.skip("Skipped!"), 'TestJit', 'test_variant_consistency_jit'),
            ),
            supports_out=False),
+    OpInfo(
+        "nn.functional.margin_ranking_loss",
+        ref=_NOTHING,
+        dtypes=all_types_and(torch.bfloat16),
```
Comment (ngimel) [inline torch/testing/_internal/common_methods_invocations.py:11820]: it's awesome that this loss is supported for integral types.

**Review by ngimel — APPROVED** [review]

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -2769,6 +2769,24 @@ def sample_inputs_randint_like(self, device, dtype, requires_grad, **kwargs):
             kwargs=sample.kwargs))
     return tuple(samples)
 
+# TODO: add reduction kwargs
+def sample_inputs_margin_ranking_loss(op_info, device, dtype, requires_grad, **kwargs):
+    _make_tensor = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
```
[the remark this answers is not in the document]
Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:2774]: People partial it however they like
